### PR TITLE
Solve bug that prevented sending StoredVmQuota and DeployedVmQuota when creating/updating Org User

### PR DESCRIPTION
**Pull request by mikeletux**
This PR is related https://github.com/vmware/terraform-provider-vcd/issues/799

## Description
This PR aims to solve a bug reported on https://github.com/vmware/terraform-provider-vcd/issues/799. When trying to update a user from whatever quota to 0 (unlimited), it wasn't working. Creating was working.

## Detailed description
VCD Terraform provider, when updating an user it was using the method `OrgUser.Update()`. Turns out that after doing some troubleshooting, I've spotted that on struct `types.User` the attributes `StoredVmQuota` and `DeployedVmQuota` had the `omitempty` tag, which was preventing to send these fields on the PUT HTTP request when updating if they were set to 0, because Go recognizes this as the empty value and doesn't sent it. 
  
The outcome of this was the API returning a 200 OK but these fields not being set to 0 (unlimited quota) because everything else was being update but those two fields, since they were not included in the request.

When creating a user it was working if set to 0, because if the API receives a POST request without these two fields, it set them to 0 (unlimited) by default.
  
To solve this, those two fields were removed that xml tag, so this fields is always sent in all PUT requests:
```
...
StoredVmQuota   int              `xml:"StoredVmQuota"`
DeployedVmQuota int              `xml:"DeployedVmQuota"`
...
```

Also `Test_UserCRUD` has been updated to test `OrgUser.Update()` and check this edge case.

### CODE REVIEW

File: types/v56/types.go [hunk 1]
```diff
@@ -2873,8 +2873,8 @@ type User struct {
 	IsExternal      bool             `xml:"IsExternal,omitempty"`
 	ProviderType    string           `xml:"ProviderType,omitempty"`
 	IsGroupRole     bool             `xml:"IsGroupRole,omitempty"`
-	StoredVmQuota   int              `xml:"StoredVmQuota,omitempty"`
-	DeployedVmQuota int              `xml:"DeployedVmQuota,omitempty"`
+	StoredVmQuota   int              `xml:"StoredVmQuota"`
+	DeployedVmQuota int              `xml:"DeployedVmQuota"`
```
[the remark this answers is not in the document]
Comment (mikeletux) [reply types/v56/types.go:2877]: Another thing would be to set `StoredVmQuota` and `DeployedVmQuota`to type *int so we have more flexibility with the omitempty tag. We would need to change some things but I think it wouldn't be very tedious 😄

[the remark this answers is not in the document]
Comment (Didainius) [reply types/v56/types.go:2877]: Should work without `*int` (unless there is any specific meaning between sending `0` and not sending at all field value at all).


One important thing to have in mind is to coordinate with PR https://github.com/vmware/terraform-provider-vcd/pull/798/

Here it changes (and fixes) the default quota from 10 to 0
https://github.com/vmware/terraform-provider-vcd/pull/798/files#diff-f598f42341eee7bfd5ce18a9daf206db9846e9bb902140c7d880a185f87cc8f0R138-R150


We need to be sure that these 2 PRs  don't give us some unexpected results when combined. Sounds like it shouldn't, but just have to notice it.

cc @adambarreiro

[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply types/v56/types.go:2877]: I would recommend checking `*int` option if there is a case of not sending any value. Quite often there is a case for that.

[the remark this answers is not in the document]
Comment (mikeletux) [reply types/v56/types.go:2877]: Not sending the value or sending the same value when updating is the exact same thing. I've tested this behavior with Postman and that's how it behaves. If I had to choose, I'd go for removing the `omitempty ` as it doesn't harm sending this field all the times.

[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply types/v56/types.go:2877]: What happens on creation - when you not send values?

**Review by lvirbalas — APPROVED** [review]
LGTM. Before merging, please converge on this discussion https://github.com/vmware/go-vcloud-director/pull/445#discussion_r827836826 with @vbauzysvmware

File: types/v56/types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikeletux) [reply types/v56/types.go:2877]: Unlimited sets by default

[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply types/v56/types.go:2877]: then should be good

**Review by vbauzysvmware — APPROVED** [review]

**Review by adambarreiro — APPROVED** [review]
